exchanges/huobi/huobi_wrapper.go

```diff
@@ -388,73 +404,73 @@ func (h *HUOBI) UpdateTickers(ctx context.Context, a asset.Item) error {
 				LastUpdated:  tt,
 			})
 			if err != nil {
-				return err
+				errs = common.AppendError(errs, err)
 			}
 		}
 	case asset.Futures:
-		linearTicks, err := h.GetBatchLinearSwapContracts(ctx)
-		if err != nil {
-			return err
+		ticks := []FuturesBatchTicker{}
+		// TODO: Linear swap contracts are coin-m assets
+		if coinMTicks, err := h.GetBatchLinearSwapContracts(ctx); err != nil {
+			errs = common.AppendError(errs, err)
+		} else {
+			ticks = append(ticks, coinMTicks...)
 		}
-		ticks, err := h.GetBatchFuturesContracts(ctx)
-		if err != nil {
-			return err
+		if futureTicks, err := h.GetBatchFuturesContracts(ctx); err != nil {
+			errs = common.AppendError(errs, err)
+		} else {
+			ticks = append(ticks, futureTicks...)
 		}
-		allTicks := make([]FuturesBatchTicker, 0, len(linearTicks)+len(ticks))
-		allTicks = append(allTicks, linearTicks...)
-		allTicks = append(allTicks, ticks...)
-		for i := range allTicks {
+		for i := range ticks {
 			var cp currency.Pair
-			if allTicks[i].Symbol != "" {
-				cp, err = currency.NewPairFromString(allTicks[i].Symbol)
-				if err != nil {
-					return err
+			var err error
+			if ticks[i].Symbol != "" {
+				cp, err = currency.NewPairFromString(ticks[i].Symbol)
+				if err == nil {
+					cp, err = h.pairFromContractExpiryCode(cp)
 				}
-				cp, err = h.convertContractShortHandToExpiry(cp, time.Now())
-				if err != nil {
-					return err
+				if err == nil {
```

Comment:
I like this `err == nil`  less when you're inspecting for errors for one of three function calls below. It becomes more ambiguous about what you're checking for and when. Its not saving many line changes and now you're nesting code that doesn't really need to be further nested

Reply:
So you're asking for:
```
M exchanges/huobi/huobi_wrapper.go
@@ -423,14 +423,15 @@ func (h *HUOBI) UpdateTickers(ctx context.Context, a asset.Item) error {
 			var cp currency.Pair
 			var err error
 			if ticks[i].Symbol != "" {
-				cp, err = currency.NewPairFromString(ticks[i].Symbol)
-				if err == nil {
-					cp, err = h.pairFromContractExpiryCode(cp)
+				if cp, err = currency.NewPairFromString(ticks[i].Symbol); err != nil {
+					errs = common.AppendError(errs, err)
+					continue
 				}
-				if err == nil {
-					cp, _, err = h.MatchSymbolCheckEnabled(cp.String(), a, true)
+				if cp, err = h.pairFromContractExpiryCode(cp); err != nil {
+					errs = common.AppendError(errs, err)
+					continue
 				}
-				if err != nil {
+				if cp, _, err = h.MatchSymbolCheckEnabled(cp.String(), a, true); err != nil {
 					if !errors.Is(err, currency.ErrPairNotFound) {
 						errs = common.AppendError(errs, err)
 					}
```
That would look like 837fd824

See, I think personally I'd double down even further on this, saving the lines in the else statement, saving another 5 lines.
My thinking on this is pretty simple:
If the issue is that we don't find the pair, then we skip the error and the tick.
It really doesn't matter where that came from.

So I'd prefer this: a1cf27f9

I've left it with the later in the branch. Shout if it's still not acceptable, obviously.